### DISCUSSION TITLE
Different description for Optional fields generated by fastapi v0.101 and v0.99

### DISCUSSION THREAD

**Discussion by aravindnujella** [fastapi/fastapi#10070] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import json

from fastapi import FastAPI
from pydantic import BaseModel

app = FastAPI()


class Item(BaseModel):
    name: str
    description: str | None = None
    price: float
    tax: float | None = None


@app.put("/items/{item_id}")
async def update_item(
    item_id: int,
    q: str | None = None,
    item: Item | None = None,
):
    results = {"item_id": item_id}
    if q:
        results.update({"q": q})
    if item:
        results.update({"item": item})
    return results


print(json.dumps(app.openapi(), sort_keys=True, indent=4))
```

### Description

FastApi behaviour for generating schemas seems to have changed from 0.88 to 0.101
In the above code, fastapi v0.101 generates `anyOf`  description for optionals specified in pydantic schema.
```
            "Item": {
                "properties": {
                    "description": {
                        "anyOf": [
                            {
                                "type": "string"
                            },
                            {
                                "type": "null"
                            }
                        ],
                        "title": "Description"
                    },
                    "name": {
                        "title": "Name",
                        "type": "string"
                    },
                    "price": {
                        "title": "Price",
                        "type": "number"
                    },
                    "tax": {
                        "anyOf": [
                            {
                                "type": "number"
                            },
                            {
                                "type": "null"
                            }
                        ],
                        "title": "Tax"
                    }
                },
                "required": [
                    "name",
                    "price"
                ],
...
```

while older version generates just the types without `anyOf`. 

```
            "Item": {
                "properties": {
                    "description": {
                        "title": "Description",
                        "type": "string"
                    },
                    "name": {
                        "title": "Name",
                        "type": "string"
                    },
                    "price": {
                        "title": "Price",
                        "type": "number"
                    },
                    "tax": {
                        "title": "Tax",
                        "type": "number"
                    }
                },
                "required": [
                    "name",
                    "price"
                ],
```

OpenAPI docs mention that by default all object properties are optional [link](https://swagger.io/docs/specification/data-models/data-types/)
> By default, all object properties are optional. You can specify the required properties in the required list:

So, it seems that `anyOf` description generated by newer fastapi is extraneous. Is there any way I can disable this?

### Operating System

macOS

### FastAPI Version

0.101.0

### Pydantic Version

2.1.1

### Python Version

3.11.4

**Comment by Kludex** [comment, marked as answer]
https://github.com/tiangolo/fastapi/pull/9873
